test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -555,25 +558,55 @@ UCS_TEST_P(test_uct_cm_sockaddr, cm_server_reject)
                 (TEST_CM_STATE_SERVER_CONNECTED | TEST_CM_STATE_CLIENT_CONNECTED)));
 }
 
-UCS_TEST_P(test_uct_cm_sockaddr, server_switch_cm)
-{
-    UCS_TEST_MESSAGE << "Testing " << m_listen_addr
-                     << " Interface: " << GetParam()->dev_name;
 
-    m_switch_cm = true;
+UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_cm_sockaddr)
 
-    ucs_async_context_t *async;
-    uct_worker_h worker_test;
-    ucs_status_t status;
 
-    status = ucs_async_context_create(UCS_ASYNC_MODE_THREAD_SPINLOCK, &async);
-    ASSERT_UCS_OK(status);
+class test_uct_cm_sockaddr_multiple_cms : public test_uct_cm_sockaddr {
+public:
+    void init() {
+        ucs_status_t status;
 
-    status = uct_worker_create(async, UCS_THREAD_MODE_SINGLE, &worker_test);
-    ASSERT_UCS_OK(status);
+        test_uct_cm_sockaddr::init();
 
-    status = uct_cm_open(GetParam()->component, worker_test, &m_cm_test);
-    ASSERT_UCS_OK(status);
+        status = ucs_async_context_create(UCS_ASYNC_MODE_THREAD_SPINLOCK, &m_async);
+        ASSERT_UCS_OK(status);
+
+        UCS_TEST_CREATE_HANDLE(uct_worker_h, m_worker_test, uct_worker_destroy,
+                               uct_worker_create, m_async,
+                               UCS_THREAD_MODE_SINGLE)
+
+        UCS_TEST_CREATE_HANDLE(uct_cm_h, m_cm_test, uct_cm_close,
+                               uct_cm_open, GetParam()->component,
+                               m_worker_test);
+    }
+
+    void cleanup() {
+        test_uct_cm_sockaddr::cleanup();
```

Comment:
should we also reset the handles? because now there would be worker which points to a destroyed m_async